api/client/credentials.go

```diff
@@ -9,36 +9,77 @@ import (
 // GetCredentials loads the user credentials from a credentials store.
 // The store is determined by the config file settings.
 func GetCredentials(c *configfile.ConfigFile, serverAddress string) (types.AuthConfig, error) {
-	s := LoadCredentialsStore(c)
+	s := LoadCredentialsStore(c, serverAddress)
 	return s.Get(serverAddress)
 }
 
 // GetAllCredentials loads all credentials from a credentials store.
 // The store is determined by the config file settings.
 func GetAllCredentials(c *configfile.ConfigFile) (map[string]types.AuthConfig, error) {
-	s := LoadCredentialsStore(c)
-	return s.GetAll()
+	ret := make(map[string]types.AuthConfig)
+	for server := range c.CredentialHelpers {
+		s := LoadCredentialsStore(c, server)
+		auths, err := s.GetAll()
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range auths {
+			ret[k] = v
+		}
+	}
+	if c.CredentialsStore != "" {
```

Comment:
These two blocks should be able to be combined by doing `LoadCredentialsStore(c, "")` and ensuring that `LoadCredentialStore` will do the right thing